api/v1beta1/common.go

```diff
@@ -28,6 +28,89 @@ import (
 // NodeSection is a specification of the node attributes
 type NodeSection struct {
 
+	// +kubebuilder:validation:Optional
+	// HostName - node name
+	HostName string `json:"hostName,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// NetworkAttachments is a list of NetworkAttachment resource names to pass to the ansibleee resource
+	// which allows to connect the ansibleee runner to the given network
+	NetworkAttachments []string `json:"networkAttachments,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// NetworkConfig - Network configuration details. Contains os-net-config
+	// related properties.
+	NetworkConfig NetworkConfigSection `json:"networkConfig"`
+
+	// +kubebuilder:validation:Optional
+	// Networks - Instance networks
+	Networks []infranetworkv1.IPSetNetwork `json:"networks,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// ManagementNetwork - Name of network to use for management (SSH/Ansible)
+	ManagementNetwork string `json:"managementNetwork,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// AnsibleHost SSH host for Ansible connection
+	AnsibleHost string `json:"ansibleHost,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// AnsibleUser SSH user for Ansible connection
+	AnsibleUser string `json:"ansibleUser,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:number"}
+	// AnsiblePort SSH port for Ansible connection
+	AnsiblePort int `json:"ansiblePort,omitempty"`
```

Comment:
nit: these coudl also be grouped

```
ansible:{
    host: "compute-1",
    port: 22,
    user : "stack"
}

i woudl be tempeted to also include SshKeysecret there too to group all the ansibel connection info to there

Reply:
+1, maybe add ansibleVars too

Reply:
Happy to split these into their own Ansible struct. Makes sense